### ISSUE TITLE
Explicit self being removed in closure argument when result assigned to var

### ISSUE THREAD

**Issue by fphilipe** [nicklockwood/SwiftFormat#511]
I'm on the latest available version 0.42.0.

Given the following Swift code

```swift
class SomeClass {
    func broken() {
        var bad = abc {
            self.foo()
            self.bar
        }

        let good = abc {
            self.foo()
            self.bar
        }
    }

    func working() {
        var bad = {
            self.bar
            self.foo()
        }

        let good = {
            self.bar
            self.foo()
        }
    }
}
```

when running `swiftformat` with the default settings, the explicit and required `self` gets stripped inside `broken()` for the `var bad` case:

```diff
 class SomeClass {
     func broken() {
         var bad = abc {
-            self.foo()
-            self.bar
+            foo()
+            bar
         }
 
         let good = abc {
             self.foo()
             self.bar
         }
     }
 
     func working() {
         var bad = {
             self.bar
             self.foo()
         }
 
         let good = {
             self.bar
             self.foo()
         }
     }
 }
```

Note that

-   the `let good` case is not affected, and that
-   in the case where the closure gets assigned directly to a `var` inside `working()`, it doesn't strip `self`.

The resulting code **does not compile**.

A workaround for now is to capture `self` in the closure under a different name:

```diff
 class SomeClass {
     func broken() {
-        var bad = abc {
-            self.foo()
-            self.bar
+        var bad = abc { [x = self]
+            x.foo()
+            x.bar
         }
 
         let good = abc {
             self.foo()
             self.bar
         }
     }
 
     func working() {
         var bad = {
             self.bar
             self.foo()
         }
 
         let good = {
             self.bar
             self.foo()
         }
     }
 }
```

**Comment by nicklockwood** [comment]
@fphilipe thanks for reporting this, I'll fix it asap.

**Comment by nicklockwood** [comment]
@fphilipe fixed in 0.43.0.

**Comment by fphilipe** [comment]
Thanks! ❤️
